packages/edit-widgets/src/components/sidebar/index.js

```diff
@@ -37,32 +37,104 @@ const WIDGET_AREAS_IDENTIFIER = 'edit-widgets/block-areas';
  */
 import WidgetAreas from './widget-areas';
 import { store as editWidgetsStore } from '../../store';
+import { unlock } from '../../lock-unlock';
+
+const { Tabs } = unlock( componentsPrivateApis );
+
+function SidebarHeader( { selectedWidgetAreaBlock } ) {
+	return (
+		<Tabs.TabList>
+			<Tabs.Tab tabId={ WIDGET_AREAS_IDENTIFIER }>
+				{ selectedWidgetAreaBlock
+					? selectedWidgetAreaBlock.attributes.name
+					: __( 'Widget Areas' ) }
+			</Tabs.Tab>
+			<Tabs.Tab tabId={ BLOCK_INSPECTOR_IDENTIFIER }>
+				{ __( 'Block' ) }
+			</Tabs.Tab>
+		</Tabs.TabList>
+	);
+}
 
-function ComplementaryAreaTab( { identifier, label, isActive } ) {
+function SidebarContent( {
+	hasSelectedNonAreaBlock,
+	currentArea,
+	isGeneralSidebarOpen,
+	selectedWidgetAreaBlock,
+} ) {
 	const { enableComplementaryArea } = useDispatch( interfaceStore );
+
+	// currentArea, and isGeneralSidebarOpen are intentionally left out from the dependencies,
+	// because we want to run the effect when a block is selected/unselected and not when the sidebar state changes.
```

Comment:
Let's add a `// eslint-disable-next-line react-hooks/exhaustive-deps` above the dependency list, and move this comment next to it for better contextual information